src/status_im/ethereum/transactions/core.cljs

```diff
@@ -145,17 +147,45 @@
 (defn get-min-known-block [db address]
   (get-in db [:wallet :accounts (eip55/address->checksum address) :min-block]))
 
+(defn min-block-transfers-count [db address]
+  (get-in db [:wallet :accounts
+              (eip55/address->checksum address)
+              :min-block-transfers-count]))
+
 (fx/defn set-lowest-fetched-block
   [{:keys [db]} address transfers]
-  (let [min-block (reduce
-                   (fn [min-block {:keys [block]}]
-                     (min (or min-block block) block))
-                   (get-min-known-block db address)
-                   transfers)]
-    {:db (assoc-in
-          db
-          [:wallet :accounts (eip55/address->checksum address) :min-block]
-          min-block)}))
+  (let [checksum (eip55/address->checksum address)
+        {:keys [min-block min-block-transfers-count]}
+        (reduce
+         (fn [{:keys [min-block] :as acc}
+              {:keys [block hash]}]
+           (cond
+             (or (nil? min-block) (> min-block block))
+             {:min-block                 block
+              :min-block-transfers-count 1}
+
+             (and (= min-block block)
+                  (nil? (get-in db [:wallet :accounts checksum :transactions hash])))
+             (update acc :min-block-transfers-count inc)
+
+             :else acc))
+         {:min-block
+          (get-min-known-block db address)
+
+          :min-block-transfers-count
+          (min-block-transfers-count db address)}
+         transfers)]
+    (log/debug "[transactions] set-lowest-fetched-block"
+               "address" address
+               "min-block" min-block
+               "min-block-transfers-count" min-block-transfers-count)
+    {:db (-> db
```

Comment:
Minor: `(update-in db [:wallet :accounts checksum]
                                      assoc :min-block min-block
                                                 :min-block-transfers-count min-block-transfers-count)`
would look better i think.